packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImplLib.sol

```diff
@@ -60,6 +65,21 @@ library ZoraCreator1155PremintExecutorImplLib {
         return abi.decode(mintArguments, (address, string));
     }
 
+    function legacySetupNewToken(address contractAddress, bytes memory encodedPremintConfig, bytes calldata signature) private returns (uint256) {
+        // for use when the erc1155 contract does not support the new delegateSetupNewToken interface, where it expects
+        // a PremintConfig as an argument.
+
+        // decode the PremintConfig from the encoded bytes.
+        PremintConfig memory premintConfig = abi.decode(encodedPremintConfig, (PremintConfig));
+
+        // call the legacy version of the delegateSetupNewToken function.
+        return ILegacyZoraCreator1155DelegatedMinter(contractAddress).delegateSetupNewToken(premintConfig, signature, msg.sender);
```

Comment:
What happens for contracts deployed pre-premint? does this just revert?

Reply:
yes they'd revert